packages/office-ui-fabric-react/src/components/TextField/TextField.base.tsx

```diff
@@ -486,57 +487,43 @@ export class TextFieldBase extends React.Component<ITextFieldProps, ITextFieldSt
     event.persist();
     const element: HTMLInputElement = event.target as HTMLInputElement;
     const value: string = element.value;
+    const props = this.props;
 
     // Avoid doing unnecessary work when the value has not changed.
-    if (value === this._latestValue) {
+    if (value === this.value) {
       return;
     }
-    this._latestValue = value;
-
-    this.setState({ value: value } as ITextFieldState, () => {
-      this._adjustInputHeight();
 
-      if (this.props.onChange) {
-        this.props.onChange(event, value);
-      }
-
-      if (this.props.onChanged) {
-        this.props.onChanged(value);
-      }
-    });
-
-    const { validateOnFocusIn, validateOnFocusOut } = this.props;
-    if (!(validateOnFocusIn || validateOnFocusOut)) {
-      this._delayedValidate(value);
+    if (props.onChange) {
+      props.onChange(event, value);
     }
 
-    if (this.props.onBeforeChange) {
-      this.props.onBeforeChange(value);
+    // ONLY is this is an uncontrolled component, update the displayed value.
+    // (Controlled components must update the `value` prop from `onChange`.)
+    if (!this._isControlled) {
+      this.setState({ uncontrolledValue: value });
```

Comment:
In theory, the new logic in componentDidUpdate should handle calling onBeforeChange and validation due to either state or props value update now (so the extra logic here isn't needed)